appengine/standard/xmpp_wikibot/wikibot.py

```diff
@@ -0,0 +1,207 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Wikibot server example using SleekXMPP client library"""
+
+import sys
+import logging
+import getpass
+from optparse import OptionParser
+from flask import Flask, request
+import requests
+import threading
+import urllib
+import json
+
+import sleekxmpp
+
+# Python versions before 3.0 do not use UTF-8 encoding
+# by default. To ensure that Unicode is handled properly
+# throughout SleekXMPP, we will set the default encoding
+# ourselves to UTF-8.
+if sys.version_info < (3, 0):
+    from sleekxmpp.util.misc_ops import setdefaultencoding
+    setdefaultencoding('utf8')
+else:
+    raw_input = input
+
+app = Flask(__name__)
+
+chat_client = None #this will be initialized when the wikibot is constructed
+
+@app.route('/send_message', methods=['GET'])
+def send_message():
+    try:
+        recipient = request.args.get('recipient')
+        message = request.args.get('message')
+    except KeyError as e:
+        logging.info("key error: {0}".format(e))
+
+    if chat_client and recipient and message:
+        chat_client.send_message(mto=recipient, mbody=message)
+        return "message sent to:" + recipient + " with body:" + message
+    else:
+        logging.info("chat client or recipient or message does not exist!")
+    return "message failed to send"
```

Comment:
Should probably return an error code as well:

    return 'message failed to send', 500